rsyslog/Sanity/imrelp-omrelp-module-test/runtest.sh

```diff
@@ -0,0 +1,218 @@
+
+        # OpenSSL certificates
+        rlRun "openssl genrsa 2048 > openssl-ca-key.pem" 0 "Generate a private key for the CA"
+        rlRun 'openssl req -new -x509 -nodes -days 365000 -key openssl-ca-key.pem -out openssl-ca-cert.pem -subj "/C=CZ/ST=Moravia/L=Brno/O=Red Hat/OU=org/CN=www.redhat.com"' 0 "Generate the X509 certificate for the CA"
+
+        rlRun 'openssl req -newkey rsa:2048 -nodes -days 365000 -keyout openssl-server-key.pem -out openssl-server-req.pem --subj "/C=CZ/ST=Moravia/L=Brno/O=Red Hat/OU=org/CN=server"' 0 "Generate the private key and certificate request for server"
+        rlRun 'openssl x509 -req -days 365000 -set_serial 01 -in openssl-server-req.pem -out openssl-server-cert.pem -CA openssl-ca-cert.pem -CAkey openssl-ca-key.pem' 0 "Generate the X509 certificate for the server"
+
+        rlRun 'openssl req -newkey rsa:2048 -nodes -days 365000 -keyout openssl-client-key.pem -out openssl-client-req.pem --subj "/C=CZ/ST=Moravia/L=Brno/O=Red Hat/OU=org/CN=client"' 0 "Generate the private key and certificate request for client"
+        rlRun 'openssl x509 -req -days 365000 -set_serial 01 -in openssl-client-req.pem -out openssl-client-cert.pem -CA openssl-ca-cert.pem -CAkey openssl-ca-key.pem' 0 "Generate the X509 certificate for the client"
+
+        rlRun 'openssl verify -CAfile openssl-ca-cert.pem openssl-ca-cert.pem openssl-server-cert.pem' 0 "Verify the server certificate"
+        rlRun 'openssl verify -CAfile openssl-ca-cert.pem openssl-ca-cert.pem openssl-client-cert.pem' 0 "Verify the client certificate"
+
+        # Store certificates
+        rlRun "mkdir -p /etc/rsyslogd.d && chmod 700 /etc/rsyslogd.d" 0 "Create /etc/rsyslogd.d"
+        rlRun "cp *.pem /etc/rsyslogd.d/"
+        rlRun "chmod 400 /etc/rsyslogd.d/* && restorecon -R /etc/rsyslogd.d"
+
+        client_config() {
+            local driver="$1"
+            rsyslogConfigReplace "SSL" <<EOF
+module(load="omrelp" tls.tlslib="$driver")
+local6.* action(type="omrelp"
+    target="127.0.0.1"
+    port="6514"
+    tls="on"
+    tls.cacert="/etc/rsyslogd.d/$driver-ca-cert.pem"
+    tls.mycert="/etc/rsyslogd.d/$driver-client-cert.pem"
+    tls.myprivkey="/etc/rsyslogd.d/$driver-client-key.pem"
+    tls.authmode="certvalid"
+    tls.permittedpeer="rsyslog")
+EOF
+            rlRun "rsyslogPrintEffectiveConfig -n"
+        }
+
+        tcfChk "config client" && {
+            rlRun "rsyslogPrepareConf"
+            rsyslogConfigAddTo "MODULES" < <(rsyslogConfigCreateSection 'SSL')
+        tcfFin; }
+
+        server_config() {
+            local driver="$1"
+            rsyslogServerConfigReplace "SSL" <<EOF
+module(
+    load="imrelp"
+    tls.tlslib="$driver"
+)
+ruleset(name="relp"){
+    action(type="omfile" file="$TmpDir/relp.log")
```

Comment:
I'm not sure if this location is ok from SELinux POV, there may be some unwanted AVCs, potentially.

Why not to use generic messages file? Of the server side, there is not much logs so there should not be any confusion. More over, the cleanup of that server's messages file is still present..

Reply:
I see also no interoperability test (gnutls<->openssl) similar to https://github.com/RedHat-SP-Security/tests/pull/37/files#diff-ce475988028d8284c64e7f551968f9949f8c304c36ca4d0472e8f6dd0578d61bR203 or https://github.com/RedHat-SP-Security/tests/pull/37/files#diff-ce475988028d8284c64e7f551968f9949f8c304c36ca4d0472e8f6dd0578d61bR220.

Did you consider also extending that gnutls-openssl test? The name would allow that, if could be just split using virtual tmt cases and a conditional phases. Bu we can merge it in future, eventually.

Reply:
At first, I was going to include it as well, however I am not entirely sure if it's possible. I used ```certtool``` to generate certificates for gnutls and ```openssl CLI``` for openssl. Right now, I can see now way how to combine them. Besides that, I think it makes more sense to use the same tls library in both server and client sides.

Reply:
> I'm not sure if this location is ok from SELinux POV, there may be some unwanted AVCs, potentially.
> 
> Why not to use generic messages file? Of the server side, there is not much logs so there should not be any confusion. More over, the cleanup of that server's messages file is still present..

I will do that.